### DISCUSSION TITLE
Need Help Trying to get ComfyUI working in Linux - Ubuntu

### DISCUSSION THREAD

**Discussion by lord-lethris** [Comfy-Org/ComfyUI#1174] [Q&A]
I Already have ComfyUI working in windows, its great - absolutely love it - but I want to be able to squeeze a bit more efficiency out of it as I'm restricted to DML and that only opens up a ridiculously small 4gb VRam.

So... I'm trying to get ComfyUI working in Ubuntu.  I've got so close, but I have got stuck at this point.
The application hangs here... I say hangs - its still responsive, but just does what seems nothing (from a front end point for view)
I have also tried suggestions I have found in google search (hence the commented out sections in the start script)
I have also tried force fp32 and fp16 and a splattering of other stuff.  but it still seems to stop doing anything here, with the following output in console:

left over keys: dict_keys(['cond_stage_model.transformer.text_model.embeddings.position_ids', 'model_ema.decay', 'model_ema.num_updates'])

![image](https://github.com/comfyanonymous/ComfyUI/assets/1736720/3c234c43-51c1-42fc-9867-a4f5873c79c1)

I've also tried different Checkpoints.
I've exhausted my knowledge of Linux and there is no definitive answer in the world of google search.  Has anyone here got any Ideas?

GPU - AMD® Radeon rx 6800 xt
CPU - AMD® Ryzen 9 5900x 12-core processor × 24
RAM - 32.0 GiB
OS - Ubuntu 22.04.3 LTS

I have installed the nightly version of Rocm (_because it got me this far, aka actually executes_) - rocm5.6
And the latest AMD drivers for Ubuntu - amdgpu-install_5.6.50600-1_all

```
linux-headers-6.2.0-26-generic is already the newest version (6.2.0-26.26~22.04.1).
amdgpu-dkms is already the newest version (1:6.1.5.50600-1609671.22.04).
amdgpu-lib is already the newest version (1:5.6.50600-1609671.22.04).
amdgpu-lib32 is already the newest version (1:5.6.50600-1609671.22.04).
rocm-developer-tools is already the newest version (5.6.0.50600-67~22.04).
rocm-hip-runtime is already the newest version (5.6.0.50600-67~22.04).
rocm-ml-libraries is already the newest version (5.6.0.50600-67~22.04).
rocm-ml-sdk is already the newest version (5.6.0.50600-67~22.04).
rocm-opencl-runtime is already the newest version (5.6.0.50600-67~22.04).
rocm-opencl-runtime set to manually installed.
rocm-opencl-sdk is already the newest version (5.6.0.50600-67~22.04).
rocm-openmp-sdk is already the newest version (5.6.0.50600-67~22.04).
rocm-utils is already the newest version (5.6.0.50600-67~22.04).
```

Any help would be grand and very welcomed and appreciated.

edit:  I'm not sure that the Checkpoint is loading - just a thought... - because in windows it pauses for a few seconds on loading checkpoint, but in Linux it speeds passed it.

**Comment by lord-lethris** [comment]
I downgraded Rocm to 5.4.2 and now I get the below when I try to render an image...
...I also created a 32gb swap partition (yea, its added to fstab) just in case as I know that was an issue on windows.  I found out that Ubuntu will install a tiny 1.5gb swap partition if you have over 16gb of RAM.  I have 32gb so my swap partition wasn't even installed.  However, that didn't help.

```
Total VRAM 16368 MB, total RAM 32002 MB
Set vram state to: NORMAL_VRAM
Device: cuda:0 AMD Radeon RX 6800 : native
Using sub quadratic optimization for cross attention, if you have memory or speed issues try using: --use-split-cross-attention
Adding extra search path checkpoints /media/lethris/Cache/ASSETS/AI-TOOLS/MODEL-ASSETS/Stable-diffusion
Adding extra search path configs /media/lethris/Cache/ASSETS/AI-TOOLS/MODEL-ASSETS/Stable-diffusion
Adding extra search path vae /media/lethris/Cache/ASSETS/AI-TOOLS/MODEL-ASSETS/VAE
Adding extra search path loras /media/lethris/Cache/ASSETS/AI-TOOLS/MODEL-ASSETS/Lora
Adding extra search path loras /media/lethris/Cache/ASSETS/AI-TOOLS/MODEL-ASSETS/LyCORIS
Adding extra search path upscale_models /media/lethris/Cache/ASSETS/AI-TOOLS/MODEL-ASSETS/ESRGAN
Adding extra search path upscale_models /media/lethris/Cache/ASSETS/AI-TOOLS/MODEL-ASSETS/RealESRGAN
Adding extra search path upscale_models /media/lethris/Cache/ASSETS/AI-TOOLS/MODEL-ASSETS/SwinIR
Adding extra search path embeddings /media/lethris/Cache/ASSETS/AI-TOOLS/MODEL-ASSETS/embeddings
Adding extra search path hypernetworks /media/lethris/Cache/ASSETS/AI-TOOLS/MODEL-ASSETS/hypernetworks
Adding extra search path controlnet /media/lethris/Cache/ASSETS/AI-TOOLS/MODEL-ASSETS/ControlNet
Starting server

To see the GUI go to: http://127.0.0.1:8188
got prompt
model_type EPS
adm 0
making attention of type 'vanilla' with 512 in_channels
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
making attention of type 'vanilla' with 512 in_channels
left over keys: dict_keys(['alphas_cumprod', 'alphas_cumprod_prev', 'betas', 'log_one_minus_alphas_cumprod', 'model_ema.decay', 'model_ema.num_updates', 'posterior_log_variance_clipped', 'posterior_mean_coef1', 'posterior_mean_coef2', 'posterior_variance', 'sqrt_alphas_cumprod', 'sqrt_one_minus_alphas_cumprod', 'sqrt_recip_alphas_cumprod', 'sqrt_recipm1_alphas_cumprod'])
./start.sh: line 9:  5745 Segmentation fault      (core dumped) python3 main.py --preview-method auto
```
![image](https://github.com/comfyanonymous/ComfyUI/assets/1736720/23e9a407-d273-4215-a93c-3e65739ea69d)

**Comment by ricperry1** [comment]
You have almost the exact same setup as me, but I haven't had even as much luck as you. I keep trying to follow AMD's ROCm instructions, but they seem way out-dated. Do you know of a better reference? I can't even get the amdgpu-install package to install. I have 5900X paired with a 6900XT and 32GB ram, and I'm using Ubuntu 22.04 LTS.
